test/unit/org/apache/cassandra/db/streaming/CassandraEntireSSTableStreamWriterTest.java

```diff
@@ -105,14 +122,23 @@ public static void defineSchemaAndPrepareSSTable()
         CompactionManager.instance.performMaximal(store, false);
 
         sstable = store.getLiveSSTables().iterator().next();
+        descriptor = sstable.descriptor;
+    }
+
+    @After
+    public void reset() throws IOException
+    {
+        // reset repair info to avoid test interfering each other
+        descriptor.getMetadataSerializer().mutateRepairMetadata(descriptor, 0, ActiveRepairService.NO_PENDING_REPAIR, false);
     }
 
     @Test
     public void testBlockWriterOverWire() throws IOException
     {
         StreamSession session = setupStreamingSessionForTest();
 
-        CassandraEntireSSTableStreamWriter writer = new CassandraEntireSSTableStreamWriter(sstable, session, CassandraOutgoingFile.getComponentManifest(sstable));
+        ComponentManifest manifest = CassandraOutgoingFile.getComponentManifest(sstable);
+        CassandraEntireSSTableStreamWriter writer = new CassandraEntireSSTableStreamWriter(sstable, session, manifest);
 
         EmbeddedChannel channel = new EmbeddedChannel();
         AsyncStreamingOutputPlus out = new AsyncStreamingOutputPlus(channel);
```

Comment:
Minor nit on the `Queue` below...could be `Queue<?>` to avoid a compiler warning.

Reply:
+1